src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -129,6 +129,11 @@ export type Fiber = Instance & {
   progressedFirstDeletion: ?Fiber,
   progressedLastDeletion: ?Fiber,
 
+  // This flag gets set to true for error boundaries when they attempt to
+  // render the error path. This way we know that if rendering fails, we should
+  // skip this error boundary as failed and propagate the error above.
+  hasErrored: boolean,
```

Comment:
I guess, the question is, what priority should the error have?

A) If no other priority can beat it, then we should just do this in a synchronous loop in place of `performDeferredWork`. If that's the case, then you can just have a separate try/catch around that one.

B) If another priority can render higher than the error set state, then this flag can be invalidated before we get around to rendering the error. Since we can still update the state of the tree before it errored.

So, I think A is probably the easiest option.